src/packages/math/src/common.js

```diff
@@ -96,12 +96,24 @@ export function atan(radians) {
   return map(radians, angle => Math.atan(angle));
 }
 
-// TODO - glsl equivalent
 export function clamp(value, min, max) {
-  return Math.max(min, Math.min(max, value));
+  return map(value, value => Math.max(min, Math.min(max, value)));
 }
 
 export function equals(a, b) {
-  return Math.abs(a - b) <=
-    glMatrix.EPSILON * Math.max(1.0, Math.abs(a), Math.abs(b));
+  if (Array.isArray(a) && Array.isArray(b)) {
+    if (a === b) {
+      return true;
+    }
+    if (a.length !== b.length) {
+      return false;
+    }
+    for (let i = 0; i < a.length; ++i) {
+      if (!equals(a[i], b[i])) {
+        return false;
+      }
+    }
+    return true;
+  }
+  return Math.abs(a - b) <= config.EPSILON * Math.max(1.0, Math.abs(a), Math.abs(b));
```

Comment:
> Technically you could just do Math.max(1, Math.abs(a))

This is the way gl-matrix does comparisons, since this library replaces gl-matrix based code it should be safe to compare this way.

The paper is interesting. I am open to changing the comparison method, though I'd prefer to see such a change accompanied by tests.